modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/StepsControlButtons.scala

```diff
@@ -143,11 +160,10 @@ object ControlButtons {
            case StopGracefullyObservation =>
              Popup("button", "Stop the current exposure at the end of the cycle")(
                Button(
-                 icon     = Some(IconStop),
                  color    = Some("orange"),
                  onClick  = requestGracefulStop(p.id, p.stepId),
-                 disabled = p.requestInFlight || p.isObservePaused
-               )
+                 disabled = p.requestInFlight || p.isObservePaused || p.nsPendingObserveCmd.isDefined
+               )(stopGracefullyIcon),
```

Comment:
I bet scalastyle complains about the trailing comma

Reply:
Yup. I changed the logic now anyway so that we can have icon groups in a nice way.